dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallback.java

```diff
@@ -328,28 +335,41 @@ public void initWithHosts(Map<Host, HostConnectionPool<CL>> hPools) {
 
 		Map<HostToken, HostConnectionPool<CL>> tokenPoolMap = new HashMap<HostToken, HostConnectionPool<CL>>();
 		
-		// Update inner state with the host tokens.
-		
+	        /* Initialize the hashtag with the first host (if hashtag is defined) */
+	        this.hashtag = allHostTokens.get(0).getHost().getHashtag();
+	        
+	        // Update inner state with the host tokens.
 		for (HostToken hToken : allHostTokens) {
+		          
+	                /**
+	                 * Checking for defined hashtags if all of them are the same.
+	                 * If not we need to throw an exception.
+	                 */
+		        String hashtagNew = hToken.getHost().getHashtag();
```

Comment:
Think through all the cases when you write code and review it. Imagine a situation like this
3 hosts:
1) hashtag = null.
2) hashtag = {}
3) hashtag = {}

What will this do?

Reply:
Will fix this.